cas_client/Cargo.toml

```diff
@@ -37,9 +37,10 @@ serde_json = { workspace = true }
 
 reqwest-middleware = "0.4"
 reqwest-retry = "0.7"
-reqwest = { version = "0.12", features = [
+reqwest = { version = "0.12.22", features = [
     "json",
     "stream",
+    "system-proxy"
```

Comment:
FYI: this doesn't look necessary. It seems to be [enabled by default](https://docs.rs/reqwest/latest/reqwest/index.html#:~:text=system%2Dproxy%20(enabled%20by%20default)%3A%20Use%20Windows%20and%20macOS%20system%20proxy%20settings%20automatically.).